### ISSUE TITLE
react-native 0.75 + expo-image build fail

### ISSUE THREAD

**Issue by RohovDmytro** [expo/expo#31029]
### Minimal reproducible example

https://-

### What platform(s) does this occur on?

Android

### Where did you reproduce the issue?

in a development build

### Summary

1. Update deps to latest
2. Try react-native 0.75.1
3. Error during build:
````
> Task :expo-image:compileDebugKotlin FAILED
e: file:///Users/rohovdmytro/Projects/houserion-debug/node_modules/expo-image/android/src/main/java/expo/modules/image/ExpoImageView.kt:221:9 Unresolved reference: resolvedLayoutDirection

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':expo-image:compileDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details
````

### Environment

 expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.4.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.12.0 - ~/.nvm/versions/node/v20.12.0/bin/node
      Yarn: 4.1.1 - ~/.nvm/versions/node/v20.12.0/bin/yarn
      npm: 10.5.0 - ~/.nvm/versions/node/v20.12.0/bin/npm
      Watchman: 2024.06.17.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
      Android SDK:
        API Levels: 23, 31, 32, 33, 34
        Build Tools: 30.0.3, 33.0.0, 33.0.1, 33.0.2, 34.0.0, 35.0.0
        System Images: android-23 | ARM 64 v8a
    IDEs:
      Android Studio: 2023.3 AI-233.14808.21.2331.11842104
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      babel-preset-expo: 11.0.14 => 11.0.14 
      expo: 51.0.28 => 51.0.28 
      expo-router: 3.5.23 => 3.5.23 
      react: ^18.3.1 => 18.3.1 
      react-dom: ^18.3.1 => 18.3.1 
      react-native: ^0.75.1 => 0.75.1 
      react-native-web: ^0.19.11 => 0.19.12 
    Expo Workflow: bare

### Expo Doctor Diagnostics

Expected package @expo/config-plugins@~8.0.0
Found invalid:
  @expo/config-plugins@3.1.0
  (for more info, run: npm why @expo/config-plugins)
Advice: Upgrade dependencies that are using the invalid package versions.

The following packages should be updated for best compatibility with the installed expo version:
  @react-native-async-storage/async-storage@1.24.0 - expected version: 1.23.1
  @react-native-community/datetimepicker@8.2.0 - expected version: 8.0.1
  @react-native-community/netinfo@11.3.2 - expected version: 11.3.1
  @shopify/flash-list@1.7.0 - expected version: 1.6.4
  react-native@0.75.1 - expected version: 0.74.5
  react-native-gesture-handler@2.18.1 - expected version: ~2.16.1
  react-native-reanimated@3.15.0 - expected version: ~3.10.1
  react-native-safe-area-context@4.10.9 - expected version: 4.10.5
  react-native-screens@3.34.0 - expected version: 3.31.1
  react-native-svg@15.5.0 - expected version: 15.2.0
  react-native-webview@13.10.5 - expected version: 13.8.6
  typescript@5.5.4 - expected version: ~5.3.3
Your project may not work correctly until you install the expected versions of the packages.
Found outdated dependencies

**Comment by rnkdsh** [comment]
same issue

**Comment by RohovDmytro** [comment]
I've tried various kotlin versions via app.config.ts, including `kotlinVersion: '2.0.10'`. Did not help :)

**Comment by Bardiamist** [comment]
Same problem

**Comment by thespacemanatee** [comment]
+1 @brentvatne, do you need a repro?

**Comment by benomzn** [comment]
Same issue

**Comment by benomzn** [comment]
Same problem, I had to downgrade react native version..

**Comment by jieey1140** [comment]
Same issue.

Has anyone found a workaround?

**Comment by RohovDmytro** [comment]
My workaround was to ditch expo-image and fallback to default react native image component by creating a custom wrapper

**Comment by thespacemanatee** [comment]
This is already [fixed](https://github.com/expo/expo/pull/30034/files#diff-41ef8ec8a2f45a84d634750b9349438b5e3bbab09a3ae4769201f984368ef83d) on `main` branch, just be patient and wait for them to release the update packages next week 😄 Though Expo should not have announced that SDK 51 works out of the box with RN 0.75.

**Comment by erickriva** [comment]
`expo-image@canary` is working fine here. Not having issues, but it's marked as canary yet for some reason.

**Comment by jingp-ofload** [comment]
+ 1 here.

**Comment by gabrieldonadel** [comment]
Hi, just a quick update, we've just released a fix for expo-image in version 1.12.14. 

Sorry for the inconvenience
